### PR TITLE
DATAUP-338: remove/replace inline font awesome styles

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Font Awesome css is included in the Jupiter stylesheet, so it does not need to be included in the JS files. This PR is step one of removing the unnecessary font awesome css inclusion.

Removing some inline font sizing for font awesome and replacing it with font awesome classes.
Had to make a bunch of corrections for eslint at the same time.

ZOMG: one set of front-end tests completed successfully!

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-338
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/subdataInput.js [hunk 1]
```diff
@@ -114,13 +127,7 @@ define([
...
         function didChange() {
             validate().then((result) => {
-                if (result.isValid) {
-                    model.setItem('value', result.value);
-                    updateInputControl('value');
-                    channel.emit('changed', {
-                        newValue: result.value,
-                    });
-                } else if (result.diagnosis === 'required-missing') {
+                if (result.isValid || result.diagnosis === 'required-missing') {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/subdataInput.js:130]: identical behaviour for `.isValid` and `diagnosis == 'required-missing'`

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/subdataInput.js [hunk 2]
```diff
@@ -250,11 +257,7 @@ define([
                                                 },
                                                 [
                                                     span({
-                                                        class: 'fa fa-minus-circle',
-                                                        style: {
-                                                            color: 'red',
-                                                            fontSize: '200%',
-                                                        },
+                                                        class: 'fa fa-2x fa-minus-circle text-danger',
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/subdataInput.js:260]: s/inline-style/font-awesome-class/

File: kbase-extension/scss/partials/_bootstrapHelper.scss [hunk 3]
```diff
@@ -5,6 +5,10 @@
     Customize bootstrap styles
 */
 
+.text-silver {
```
Comment (ialarmedalien) [inline kbase-extension/scss/partials/_bootstrapHelper.scss:8]: bootstrap has a load of `text-(danger|success|primary|warning)` colours -- this adds silver to that set.

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/subdataInput.js [hunk 4]
```diff
@@ -39,22 +38,36 @@ define([
...
-        let spec = config.parameterSpec,
+        const spec = config.parameterSpec,
             runtime = Runtime.make(),
             workspaceId = runtime.getEnv('workspaceId'),
             busConnection = runtime.bus().connect(),
             channel = busConnection.channel(config.channelName),
             paramsChannel = busConnection.channel(config.paramsChannelName),
-            parent,
+            subdataMethods = SubdataMethods.make(),
+            model = Props.make({
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/subdataInput.js:48]: from bottom of file

**Review by briehl — CHANGES_REQUESTED** [review]
Looks good. Found one little typo, maybe?

**Pull request merged**